src/metabase/query_processor/pivot.clj

```diff
@@ -119,31 +165,20 @@
    its application via `partial`.
 
    You are expected to wrap this call in `qp.streaming/streaming-response` yourself."
+  ([query]
+   (run-pivot-query query nil))
+  ([query info]
+   (run-pivot-query query info (context.default/default-context)))
   ([query info context]
-   (qp.store/with-store
-     (let [main-breakout           (:breakout (:query query))
-           col-determination-query (add-grouping-field query main-breakout 0)
-           all-expected-cols       (qp/query->expected-cols col-determination-query)
-           all-queries             (generate-queries query)]
-       (process-multiple-queries
-        all-queries
-        (assoc context
-               :info (assoc info :context context)
-               ;; this function needs to be executed at the start of every new query to
-               ;; determine the mapping for maintaining query shape
-               :column-mapping-fn (fn [query]
-                                    (let [query-cols (map-indexed vector (qp/query->expected-cols query))]
-                                      (map (fn [item]
-                                             (some #(when (= (:name item) (:name (second %)))
-                                                      (first %)) query-cols))
-                                           all-expected-cols)))
-               ;; this function needs to be called for each row so that it can actually
-               ;; shape the row according to the `:column-mapping-fn` above
-               :row-mapping-fn (fn [row context]
-                                 ;; the first query doesn't need any special mapping, it already has all the columns
-                                 (if-let [col-mapping (:pivot-column-mapping context)]
-                                   (map (fn [mapping]
-                                          (when mapping
-                                            (nth row mapping)))
-                                        col-mapping)
-                                   row))))))))
+   ;; make sure the query keys are normalized to lisp-case.
+   (let [query (normalize/normalize query)]
+     (qp.store/with-store
```

Comment:
code will still work with `snake_case` keys (not sure if this actually is needed)